src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -1426,6 +1427,13 @@ public <V extends View> V configRoundtrip(V view) throws Exception {
         return view;
     }
 
+    public <C extends Cloud> C configRoundtrip (C cloud) throws Exception {
+        jenkins.clouds.add(cloud);
```

Comment:
Does this replace any existing `Cloud` of the same `name`?

Reply:
Under the hood, the cloud object is stored in a `ArrayList` and it should not overwrite an already existing element in the list.

Reply:
Then I would suggest one of
* checking for an existing `Cloud` of the same `name` and deleting it first, so you are actually replacing it
* removing this `add` & `save`, so the caller (test case) is responsible for registering the `Cloud` initially
* documenting the current behavior, since if I recall correctly it differs from that of other `configRoundtrip` overloads